users/dshields/dshields.c

```diff
@@ -2,7 +2,7 @@
 #include "dshields.h"
 
 bool process_record_user(uint16_t keycode, keyrecord_t *record) {
-    if (keycode == KC_ESC && record->event.pressed) {
+    if ((keycode == KC_ESC || keycode == MT_ESC) && record->event.pressed) {
```

Comment:
You may want something like this: 
```c
    if ((keycode >= QK_MOD_TAP && keycode <= QK_MOD_TAP_MAX) || (keycode >= QK_LAYER_TAP && keycode <= QK_LAYER_TAP_MAX) || (keycode >= QK_MODS && keycode <= QK_MODS_MAX)) {
        keycode = keycode & 0xFF;
    }
```

This will extract the keycode from mod tap keys, layer tap keys, and modded keys.   This way, you don't have to check for each, but will get the "ESC" key, no matter what.

Reply:
Thanks, I've incorporated the changes you've suggested.